tests/Signals_perftest2.m

```diff
@@ -0,0 +1,161 @@
+    BasicOps = {@gt, @ge, @lt, @le, @eq,...
+        @plus, @minus, @times, @rdivide} % logical/arithmetical subset of functions to test
+  end
+  
+  methods (TestMethodSetup)
+    
+    function createNetwork(testCase)
+      % creates the signals network in which we will run the tests
+      testCase.Net = sig.Net;
+      testCase.A = testCase.Net.origin('A');
+      testCase.B = testCase.Net.origin('B');
+    end
+    
+  end
+    
+  methods (Test)
+    
+    function test_singleOps(testCase, Ops)
+      % benchmarks operations on a single signal without propogations
+      
+      % set-up op to test
+      switch func2str(Ops)
+        % logical and scalar ops
+        case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times', 'rdivide'}
+          c = feval(Ops, testCase.A, testCase.B); %#ok<*NASGU>
+          post(testCase.B, 2);
+        % listener ('onValue') & 'map' ops
+        case {'onValue', 'map'}
+          f = @plus;
+          c = feval(Ops, testCase.A, @(x) f(1,x));
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'scan'
+        case {'scan'}
+          f = @plus;
+          c = feval(Ops, testCase.A, f, testCase.B);
+          post(testCase.B, 0);
+          while (testCase.keepMeasuring)
+            post(testCase.A, 1);
+          end
+        % 'subscriptable'
+        case {'subscriptable'}
+          s = struct('A', 1);
+          c = feval(Ops, testCase.A);
+          c_A = c.A;
+          while (testCase.keepMeasuring)
+            post(testCase.A, s);
+          end
+      end
+      
+      % test op
+      switch func2str(Ops)
+        case {'nop', 'post'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              feval(Ops, testCase.A, 1);
+            end
+          end
+        case {'gt', 'ge', 'lt', 'le', 'eq', 'plus', 'minus', 'times',...
+            'rdivide', 'onValue', 'map', 'scan'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, 1); % this post will trigger the update of signal 'c', defined above
+            end
+          end
+        case {'subscriptable'}
+          while (testCase.keepMeasuring)
+            for i = 1:testCase.Reps
+              post(testCase.A, s); % this post will trigger the update of signal 'c_A', defined above
+            end
+          end
+      end
+      
+    end
+    
+    function test_networkOps(testCase, OnlyBasicOps, NumNodes)
+      % benchmarks operations on a signals network upon posting to a signal
+      % which updates all other signals in the network
+      
+      % create an empty array that will grow
+      vals = [];
+      
+      % set-up network to test
+      otherSigs = cell(NumNodes,1);
+      if OnlyBasicOps % pick operation for signal at random amongst 'testCase.BasicOps'
```

Comment:
I wouldn't make this random: you want the performance test to be consistent and reproducible, otherwise it's impossible to determine how variable the performance truly is.

Reply:
true